guest/enqueue/enqueue.go

```diff
@@ -0,0 +1,65 @@
+/*
+   Copyright 2023 The Kubernetes Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// Package enqueue is defined internally so that it can export Pod as
+// cyclestate.Pod, without circular dependencies or exporting it publicly.
+package enqueue
+
+import (
+	"runtime"
+	"unsafe"
+
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/api"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/plugin"
+)
+
+// enqueue is the current plugin assigned with SetPlugin.
+var enqueue api.EnqueueExtensions
+
+// SetPlugin is exposed to prevent package cycles.
+func SetPlugin(enqueueExtensions api.EnqueueExtensions) {
+	if enqueueExtensions == nil {
+		panic("nil enqueueExtensions")
+	}
+	enqueue = enqueueExtensions
+	plugin.MustSet(enqueueExtensions)
+}
+
+// prevent unused lint errors (lint is run with normal go).
+var _ func() = _enqueue
+
+// enqueue is only exported to the host.
+//
+//export enqueue
+func _enqueue() {
```

Comment:
currently enqueue returns nothing, not a framework status. We can change it if this does. I chose the name enqueue as it is a lot simpler than `EventsToRegister`. However, if there is another function for enqueue, let me know and I can change it...